packages/ragbits-cli/src/ragbits/cli/app.py

```diff
@@ -0,0 +1,76 @@
+import json
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any
+
+import typer
+from pydantic import BaseModel
+from rich.console import Console
+from rich.table import Table
+
+
+class OutputType(Enum):
+    """Indicates a type of CLI output formatting"""
+
+    TEXT = "text"
+    JSON = "json"
+
+
+@dataclass()
+class CliState:
+    """A dataclass describing CLI state"""
+
+    output_type: OutputType = OutputType.TEXT
+
+
+class CLI(typer.Typer):
+    """A CLI class with output formatting"""
+
+    def __init__(self, *args: Any, **kwargs: Any):  # noqa: ANN401
+        super().__init__(*args, **kwargs)
+        self.state: CliState = CliState()
+        self.console: Console = Console()
+
+    def set_output_type(self, output_type: str) -> None:
+        """
+        Set the output type in the app state
+        Args:
+            output_type: str
+        Raises:
+            ValueError - if the output_type is not `json` or `text`
+        """
+        if output_type not in ["text", "json"]:
+            raise ValueError("Output type must be either 'text' or 'json'")
+        if output_type == "json":
+            self.state.output_type = OutputType.JSON
+        if output_type == "text":
+            self.state.output_type = OutputType.TEXT
```

Comment:
```suggestion
        Raises:
            ValueError - if the output_type is not one of the values in `OutputType`
        """
        self.state.output_type = OutputType(output_type)
```

Reply:
(although typing the callback with the `OutputType` enum will mean that `set_output_type` cal take an anum as well, making the problem go away altogether)

Reply:
done - callback typed with the OutputType